pkg/apis/serving/v1alpha1/kfservice_framework_tensorrt.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 kubeflow.org.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"strings"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	defaultTensorRTISImageName = "nvcr.io/nvidia/tensorrtserver"
+	// For versioning see https://github.com/NVIDIA/tensorrt-inference-server/releases
+	defaultTensorRTISRuntimeVersion = "19.05-py3"
+	// InvalidModelURIError issue fix is tracked by https://github.com/kubeflow/kfserving/issues/148
+	InvalidModelURIError = "Model URI must be prefixed by gs:// (only Google Cloud Storage paths are supported for now."
+	TensorRTISGRPCPort   = int32(9000)
+	TensorRTISRestPort   = int32(8080)
+)
+
+func (t *TensorRTSpec) CreateModelServingContainer(modelName string, config *FrameworksConfig) *v1.Container {
+	imageName := defaultTensorRTISImageName
+	if config.TensorRT.ContainerImage != "" {
+		imageName = config.TensorRT.ContainerImage
+	}
+
+	// based on example at: https://github.com/NVIDIA/tensorrt-laboratory/blob/master/examples/Deployment/Kubernetes/basic-trtis-deployment/deploy.yml
+	return &v1.Container{
+		Image:     imageName + ":" + t.RuntimeVersion,
+		Resources: t.Resources,
+		Args: []string{
+			"trtserver",
+			"--model-store=" + t.ModelURI,
+			"--allow-poll-model-repository=false",
+			"--allow-grpc=true",
+			"--allow-http=true",
+			"--grpc-port=" + fmt.Sprint(TensorRTISGRPCPort),
+			"--http-port=" + fmt.Sprint(TensorRTISRestPort),
+		},
+		Ports: []v1.ContainerPort{
+			v1.ContainerPort{
+				ContainerPort: TensorRTISRestPort,
+			},
+		},
+	}
+}
+
+func (t *TensorRTSpec) ApplyDefaults() {
+	if t.RuntimeVersion == "" {
+		t.RuntimeVersion = defaultTensorRTISRuntimeVersion
+	}
+	setResourceRequirementDefaults(&t.Resources)
+}
+
+func (t *TensorRTSpec) Validate() error {
```

Comment:
Need to check RuntimeVersion

Reply:
Dont have an acceptable versions list yet, so skipping for now

Reply:
Not sure what to validate it to... have a default, so not setting it is fine